### PR TITLE
Add ELF encoding detection

### PR DESCRIPTION
**Pull request by angrygoats**
@a-morales @rschifflin @acmcarther

Quick PR to add ELF encoding detection based on the E_IDENT block. I'm considering replacing indicies used to navigate the E_IDENT block - looking for opinions on that.

### CODE REVIEW

**Comment by a-morales** [comment]
:+1:

File: src/formats/elf.rs [hunk 1]
```diff
@@ -89,7 +102,14 @@ pub fn get_version(elf_header: &ElfHeader) -> Result<ElfVersion, String> {
     }
 }
 
-//pub fn get_data_encoding
+pub fn get_data_encoding(elf_header: &ElfHeader) -> Result<ElfEncoding, String> {
```
Comment (acmcarther) [inline src/formats/elf.rs]: You could define this as a static method on ELFHeader, like `as_elf_encoding`. See [the convention] https://aturon.github.io/style/naming.html#conversions.

Comment (acmcarther) [inline src/formats/elf.rs]: Same with the other conversions in this file.

File: tests/formats.rs [hunk 2]
```diff
@@ -128,8 +130,23 @@ mod tests {
                 }
             }
 
-            it "returns the data encoding" {
+            describe! encoding {
+                it "detects invalid encodings" {
+                    assert_eq!(get_elf_data_encoding(&elf_header_1).unwrap(), ElfEncoding::ElfDataNone);
```
Comment (acmcarther) [inline tests/formats.rs]: :+1:

**Comment by acmcarther** [comment]
One optional comment, LGTM

**Pull request closed without merging**